### PR TITLE
Updated fixture methods inheritance explanation

### PR DESCRIPTION
**Pull request by anshbansal**
Added that "Fixture methods overridden in subclasses will be executed like class constructors."

### CODE REVIEW

**Comment by anshbansal** [comment]
This for the issue https://github.com/spockframework/spock/issues/94

File: docs/spock_primer.adoc [hunk 1]
```diff
@@ -98,7 +98,7 @@ Usually it's a good idea to use a fresh fixture for every feature method, which
 methods are for. Occasionally it makes sense for feature methods to share a fixture, which is achieved by using shared
 fields together with the `setupSpec()` and `cleanupSpec()` methods. All fixture methods are optional.
 
-Note: The `setupSpec()` and `cleanupSpec()` methods may not reference instance fields.
+Note: The `setupSpec()` and `cleanupSpec()` methods may not reference instance fields. Fixture methods overridden in subclasses will be executed like class constructors.
```
Comment (leonard84) [inline docs/spock_primer.adoc]: Could you expand/reword the explanation to make it clearer what is happening. That fixture methods cannot be overridden and will always execute from parent to child.

Comment (robfletcher) [inline docs/spock_primer.adoc]: Also I think it's confusing likening them to constructors

**Comment by anshbansal** [comment]
Should we also update the part starting with "Conceptually, a feature method consists of four phases". It may become too verbose.

**Comment by leonard84** [comment]
No, I would keep that part as it is now.

**Comment by anshbansal** [comment]
So @leonard84  / @robfletcher  can you merge this? I have already updated the explanation.

**Comment by anshbansal** [comment]
bump

**Comment by anshbansal** [comment]
bump

**Comment by robfletcher** [comment]
Merged in 48f2be135957eaeec98c5cde114833b192a37d79

**Pull request closed without merging**